algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -108,32 +152,45 @@ class QTreeSemigroup[A](k: Int)(implicit val underlyingMonoid: Monoid[A]) extend
   }
 }
 
-case class QTree[A](
+case class QTree[@specialized(Int, Long, Float, Double, Unit) A] private[algebird] (
   offset: Long, //the range this tree covers is offset*(2^level) ... (offset+1)*(2^level)
   level: Int,
   count: Long, //the total count for this node and all of its children
   sum: A, //the sum at just this node (*not* including its children)
-  lowerChild: Option[QTree[A]],
-  upperChild: Option[QTree[A]]) {
+  private[algebird]lowerChildNullable: QTree[A],
```

Comment:
Okay. I didn't know that. Related to my comments on case class with private constructors.